flow/shared/authenticator.go

```diff
@@ -0,0 +1,218 @@
+package shared
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type AuthenticationConfig struct {
```

Comment:
nit: don't think we need the struct tags
also could be a proto for consistency

Reply:
Kept it like this to match upstream, if we keep it coming from environment we may get rid of it

Want to get feedback from Pete if these split up environment variables work or if infra will prefer the json as a single variable or if we go as far as parsing it. Hope this iteration sticks tho